database/insertData/dev/03_template_G_basicRenewalTesting.js

```diff
@@ -0,0 +1,188 @@
+                      { minute: 5 }
+                    ]
+                  }
+                  matchField: "serial"
+                  matchValue: {
+                    # ugly hack to get value from array
+                    type: "string",
+                    operator: "+",
+                    children: [
+                      {
+                        operator: "objectProperties",
+                        children: [
+                          "applicationData.responses.Q1.selection.serial"
+                        ]
+                      },
+                      ""
+                    ]
+                  }
+                }
+              }
+              {
+                actionCode: "scheduleAction"
+                trigger: ON_APPLICATION_SUBMIT
+                sequence: 102
+                parameterQueries:{
+                  eventCode: "prod_exp1"
+                  duration: { minute: 10 }
+                }
+              }
+              {
+                actionCode: "sendNotification"
+                trigger: ON_SCHEDULE
+                eventCode: "prod_exp1"
+                condition: {
+                  operator: "=",
+                  children: [
+                    {
+                      operator: "graphQL",
+                      children: [
+                        "query CheckExpired($id: Int!) {products(filter: {id: {equalTo: $id}, expiryDate: {lessThan: \\"now()\\"}}) {totalCount}}",
+                        "graphQLEndpoint",
+                        [
+                          "id"
+                        ],
+                        {
+                          operator: "objectProperties",
+                          children: [
+                            "outputCumulative.product.id"
+                          ]
+                        },
+                        "products.totalCount"
+                      ]
+                    },
+                    1
+                  ]
+                }
+                parameterQueries: {
+                  fromName: "Application Manager"
+                  fromEmail: "no-reply@sussol.net"
+                  email: {
+                    operator: "objectProperties"
+                    children: ["applicationData.email", ""]
+                  }
+                  subject: {
+                    operator: "stringSubstitution"
+                    children: [
+                      "Your product %1 has expired",
+                      {
+                        operator: "objectProperties"
+                        children: [ "outputCumulative.product.name", ""]
+                      }
+                    ]
+                  }
+                  message: "Your product registration has expired. If you wish to continue, you'll need to re-apply.\\n(Note: this is just a test to check the notification worked. Nothing proper has actually changed)"
+                }
+              }
+            ]
+          }
```

Comment:
> change the product to EXPIRED at the same time the notification for "Your product registration has expired" is sent

No, there's nothing to set here. The product is expired because it's "expiry_date" field is in the past. (When it's renewed it gets set to the future again)

Reply:
> I'm trying to understand how this is working - so before you had something very similar:
> 
>     * An sendNotification that would run after 5 minutes when this product was expired
> 
>     * After submitting this application you would increase the expiry date for 5 minutes more (done in the first action `scheduleAction`
> 
>     * Then there is another `sendNotification` setup to notify user that product has expired that would run after this new 5 minutes have passed
> 
> 
> So the first `sendNotification` would have already run - and the condition wouldn't match - expiry is above that period and then it would try to run again? Now we have a new `sendNotification` that should only run when the expiry date is over and it would send the message just in case it hasn't been renewed?

I'm not fully understanding what you're saying there, but basically -- the "renewal" application is intended to work pretty much the same as the "review" one as far as expiry goes. Please come and chat to me to explain what you're asking? (Or suggesting?)

Reply:
Hmmm ok. That sounds goo.

I guess that the other point raised (on first comment) is resolved with what I was reading after over this PR. When you "create" one action using the same code for an existing eventCode, it would keep the same action and just update the "when to run" field. Correct?

If that's the case, I guess if it would be nice to have a simpler way to define this **updated** action....